opm/autodiff/StandardWell.hpp

```diff
@@ -57,15 +57,30 @@ namespace Opm
         using Base::has_polymer;
         using Base::has_energy;
 
-        // the positions of the primary variables for StandardWell
-        // there are three primary variables, the second and the third ones are F_w and F_g
-        // the first one can be total rate (G_t) or bhp, based on the control
+        // polymer concentration and temperature are already known by the well, so
+        // polymer and energy conservation do not need to be considered explicitly
+        static const int numPolymerEq = has_polymer ? 1 : 0;
+        static const int numEnergyEq = has_energy ? 1 : 0;
+        static const int numWellEq = numEq + 1 - numPolymerEq - numEnergyEq;
 
+        // the positions of the primary variables for StandardWell
+        // the first one is the weighted total rate (G_t), the second and the third ones are F_w and F_g,
+        // which represent the fraction of Water and Gas based on the weighted total rate, the last one is BHP.
+        // correspondingly, we have four well equations for blackoil model, the first three are mass
+        // converstation equations, and the last one is the well control equation.
+        // primary variables related to other components, will be before the Bhp and after F_g.
+        // well control equation is always the last well equation.
+        // TODO: in the current implementation, we use the well rate as the first primary variables for injectors,
+        // instead of G_t.
         static const bool gasoil = numEq == 2 && (Indices::compositionSwitchIdx >= 0);
-        static const int XvarWell = 0;
+        static const int GTotal = 0;
         static const int WFrac = gasoil? -1000: 1;
```

Comment:
Do we have a more descriptive name for this (these) variable?
Could this work?
```
GTotal -> weightedTotalRate
WFrac -> waterFraction etc. 
```

Reply:
It will work. In my opinion, they are frequently used indices, and kind of good to keep them short.  It is okay to give comments about their definition when defining them. 

Anyway, I do not have strong feeling about this. If you insist, I will do a search-replace through the file, or in another PR. 

Just let me know how you think.

Reply:
I see your point. No strong opinions on my side either. But what is the "G" in GTotal?

Reply:
I do not recall the reason, probably I felt `Q` is not very correct since it is not the total well rate, then I used `G`. Maybe I should use `WQTotal` to say it is `weighted Q total`.